src/installer/pkg/sfx/installers/host.wxs

```diff
@@ -47,6 +62,16 @@
     <DirectoryRef Id="$(var.Program_Files)">
       <Directory Id="PROGRAMFILES_DOTNET" Name="dotnet" />
     </DirectoryRef>
+    
+    <?if $(var.Platform)~=x64 ?>
+    <CustomActionRef Id="Set_PROGRAMFILES_DOTNET_NON_NATIVE_ARCHITECTURE" />
```

Comment:
I am not too familiar with WiX ordering - is this guaranteed to run before the components' CopyFile?

Reply:
This just references the authoring which includes the custom action.  The ordering for that is defined in the SetProperty action.  It is scheduled to happen after Set_NON_NATIVE_ARCHITECTURE which happens before CostFinalize https://github.com/dotnet/arcade/blob/0ec8dbacaefa7ccdb4bc96e120e07dc60c6eda98/src/Common/wix/dotnethome_x64.wxs#L23, which happens well before `InstallFiles` https://docs.microsoft.com/en-us/windows/win32/msi/suggested-installexecutesequence.   `CostFinalize` is the action that applies properties to directories